packages/shared/src/components/MainLayout.tsx

```diff
@@ -87,6 +100,17 @@ function MainLayout({
     setOpenMobileSidebar,
   });
 
+  useEffect(() => {
+    if (pathname !== previousPathname) {
+      const newFeedName = getFeedName(pathname, {
```

Comment:
Are you sure `getFeedName` returns the same thing as `ActiveFeedContext.feedName` used to? Because that one used to return value of `AllFeedPages` and this computes the feed from path which is not always the same. 

For example if we are on the https://app.daily.dev/squads/ai `getFeedName` will return `squads[handle]` but the `feedName` passed to `Feed` component is `squad`.

Reply:
This is the issue raised in that Decision record where we decided to move on like this. This method is actually the one used in ActiveFeedContext to get [feedName](https://github.com/dailydotdev/apps/blob/4289af7ec8b1a7842e54f3402680cdd63d6a3eda/packages/shared/src/components/MainFeedLayout.tsx#L119), but on the code still there are further modifications before it reaches ActiveFeedContext.

At the moment, for the feeds we need it, it does work ok, but can't reassure it will work as should for any newer use cases. 
I can also create a new func to check feedname for the feeds we need it so we are more secure.

Reply:
> This method is actually the one used in ActiveFeedContext to get [feedName](https://github.com/dailydotdev/apps/blob/4289af7ec8b1a7842e54f3402680cdd63d6a3eda/packages/shared/src/components/MainFeedLayout.tsx#L119),

I don't think it is, `ActiveFeedContext` used to use `feedName` props from `Feed`, look https://github.com/dailydotdev/apps/blob/3fab72ca2ebe77b2e258436824532af59e18f4b8/packages/shared/src/components/Feed.tsx#L148-L154

Its not that your logic is invalid but the change of where you pull `feedName` from results in different values.

Reply:
Yeah true, the example I gave above is how the feedName prop is generated in one of the places where <Feed ..... /> is rendered with that feedName prop

Reply:
Yeah, the main issue seems to be we have a prop `Feed.feedName` and a function `getFeedName` that look to be the same but are completely different things. Not sure what is the best to solve it at this point but for now most important thing is to validate non of other logic breaks.

Reply:
For sure, we're using the new context with feedName only for evaluation on the feed layout experiment, none of the previous logic has been changed.